client/state/reader/site-blocks/reducer.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * Internal dependencies
+ */
+import {
+	READER_SITE_BLOCK_REQUEST_SUCCESS,
+	READER_SITE_UNBLOCK_REQUEST_SUCCESS,
+} from 'state/action-types';
+import { createReducer, keyedReducer } from 'state/utils';
+
+/**
+ * Tracks all known site block statuses, indexed by site ID.
+ */
+export const items = keyedReducer( 'siteId', createReducer( {}, {
+	[ READER_SITE_BLOCK_REQUEST_SUCCESS ]: ( state, action ) => action.data.success,
```

Comment:
should we also set the block / unblock optimistically, prior to the api returning? otherwise we'll need a thing to indicate progress...